File: Conditional_Routing/nodejs/device_type.js
const MOBILE_PREFIX = '/mobile';
const TABLET_PREFIX = '/tablet';
const MOBILE_REG = /phone/i;
const TABLET_REG = /pad/i;

async function f(event) {
    const request = event.request;
    const headers = request.headers;

    let prefix = '';
    if (headers['user-agent']) {
        // 利用正则匹配不同 UA
        if (MOBILE_REG.test(headers['user-agent'])) {
            prefix = MOBILE_PREFIX;
        } else if (TABLET_REG.test(headers['user-agent'])) {
            prefix = TABLET_PREFIX;
        }
    }

    // 更新 uri
    request.uri = prefix + request.uri;

    // 自动回源
    return request;

}

exports.handler = f;

File: Conditional_Routing/python/device_type.py
import re

MOBILE_PREFIX = '/mobile'
TABLET_PREFIX = '/tablet'
MOBILE_REG = '.*phone.*'
TABLET_REG = '.*pad.*'


async def handler(event):
    request = event.request
    headers = request.headers

    prefix = ''
    if 'user-agent' in headers:
        # 利用正则匹配不同 UA
        ua = headers.get('user-agent')
        if re.match(MOBILE_REG, ua, re.I):
            prefix = MOBILE_PREFIX
        elif re.match(TABLET_REG, ua, re.I):
            prefix = TABLET_PREFIX

    # 更新 uri
    request.uri = prefix + request.uri

    # runtime 回源
    return Request(request=request)
